### PR TITLE
Removing inverse synths

### PR DESCRIPTION
**Pull request by justinjmoses**
In preparation for SIP-120 removing `ExchangeRates` (#1127) it's prudent to simply remove inverse pricing altogether.

### CODE REVIEW

**Review by dbeal-eth — APPROVED** [review]
a lot is gone

**Review by barrasso — APPROVED** [review]

File: contracts/PurgeableSynth.sol [hunk 1]
```diff
@@ -51,10 +51,7 @@ contract PurgeableSynth is Synth {
         uint maxSupplyToPurge = exRates.effectiveValue("sUSD", maxSupplyToPurgeInUSD, currencyKey);
 
         // Only allow purge when total supply is lte the max or the rate is frozen in ExchangeRates
```
Comment (artdgn) [inline contracts/PurgeableSynth.sol:53]: nit: does the comment and revert message need updating too?

File: contracts/ExchangeRates.sol [hunk 2]
```diff
@@ -96,71 +87,6 @@ contract ExchangeRates is Owned, MixinSystemSettings, IExchangeRates {
...
-            inverse.frozenAtUpperLimit = false;
-            inverse.frozenAtLowerLimit = false;
-            // remove any tracking
-            roundFrozen[currencyKey] = 0;
-        }
-
-        // SIP-78
-        uint rate = _getRate(currencyKey);
-        if (rate > 0) {
-            exchanger().setLastExchangeRateForSynth(currencyKey, rate);
```
Comment (artdgn) [inline contracts/ExchangeRates.sol:145]: Can `Exchanger.setLastExchangeRateForSynth` be removed now as well?

Comment (justinjmoses) [reply contracts/ExchangeRates.sol:145]: good catch

File: contracts/PurgeableSynth.sol (hunk 1 above)
Comment (justinjmoses) [reply contracts/PurgeableSynth.sol]: fixed both

**Review by artdgn — APPROVED** [review]

File: test/contracts/SynthUtil.js [hunk 3]
```diff
@@ -2,10 +2,7 @@
 
 const { contract } = require('hardhat');
 const { assert, addSnapshotBeforeRestoreAfterEach } = require('./common');
-const {
-	toBytes32,
-	constants: { ZERO_BYTES32 },
-} = require('../..');
+const { toBytes32 } = require('../..');
 const { toUnit, currentTime } = require('../utils')();
 const { setExchangeFeeRateForSynths } = require('./helpers');
```
Comment (sohkai) [inline test/contracts/SynthUtil.js:7]: Just wondering, is it worth removing the `iBTC` bits in this file as well? For example, l51 still sets the `iBTC` rate on ExchangeRates.

File: test/contracts/PurgeableSynth.js [hunk 4]
```diff
@@ -290,7 +290,6 @@ contract('PurgeableSynth', accounts => {
 				});
 
 				describe('when the user holds 5000 USD worth of the purgeable synth iETH', () => {
```
Comment (sohkai) [inline test/contracts/PurgeableSynth.js:292]: Similar to above, worth removing `iETH` bits altogether here?

This test also seems shallow now; will holders of isynths not be able to invoke a purge from here on?

File: publish/src/commands/deploy/deploy-core.js [hunk 5]
```diff
@@ -57,7 +57,6 @@ module.exports = async ({
 
 	await deployer.deployContract({
 		name: 'ExchangeRates',
-		source: useOvm ? 'ExchangeRatesWithoutInvPricing' : 'ExchangeRates',
```
Comment (sohkai) [inline publish/src/commands/deploy/deploy-core.js:60]: For SIP-120 we might need to keep / re-add this during deployment to switch between the base ExchangeRates and the one with dex pricing

**Review by sohkai — COMMENTED** [review]
🙌 not totally up to speed on the status of isynths and if any are still left in wallets, but I'm assuming there aren't any. Looks good!

Just noted a couple of tests that still had isynths (not an exhaustive search though).

**Review by jacko125 — APPROVED** [review]
changes LGTM

File: test/contracts/PurgeableSynth.js (hunk 4 above)
Comment (justinjmoses) [reply test/contracts/PurgeableSynth.js:292]: correct. iSynths will never be purged - we've used the `SynthRedeemer` for deprecated ones and no new iSynths will be created.

File: publish/src/commands/deploy/deploy-core.js (hunk 5 above)
Comment (justinjmoses) [reply publish/src/commands/deploy/deploy-core.js:60]: Yep, going to update 120 with this change.

**Pull request merged**